examples/double_gyre.jl

```diff
@@ -0,0 +1,307 @@
+# # Double Gyre
+#
+# This example simulates a double gyre following:
+# https://mitgcm.readthedocs.io/en/latest/examples/baroclinic_gyre/baroclinic_gyre.html
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: on_architecture
+using Oceananigans.TurbulenceClosures: VerticallyImplicitTimeDiscretization
+using JLD2
+using CairoMakie
+using Statistics
+using Printf
+
+const Lλ = 40.0 # [°] longitude extent of the domain
+const Lφ = 60.0 # [°] latitude extent of the domain
+const Lz = 1.8kilometers # depth [m]
+const φ₀ = 15.0 # [°N] latitude of the center of the domain
```

Comment:
I was thinking that it may be cleaner if we define the domain via, e.g., 

```suggestion
const λ_west = -30 # [°] longitude of west boundary
const λ_west = +30 # [°] longitude of east boundary
const φ_south = 15 # [°] latitude of south boundary
const φ_north = 75 # [°] latitude of north boundary
```

What do others think?

Reply:
I like it!

Reply:
Go for it!